src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -535,16 +532,25 @@ private void Resize(int newSize, bool forceNewHashCodes)
                 }
             }
 
+            int k = 0;
             for (int i = 0; i < count; i++)
             {
-                if (entries[i].hashCode >= 0)
+                if (_entries[i].hashCode >= 0)
                 {
-                    int bucket = entries[i].hashCode % newSize;
-                    entries[i].next = buckets[bucket];
+                    ref Entry entry = ref _entries[i];
```

Comment:
Caching ` ref Entry entry = ref _entries[i];` in a local before the if-check and using it in the if-check would be better, I think.

Reply:
The JIT tends to handle well repeated accesses to the same array element so caching element refs may not do much. What might help in this this kind of code is caching the array itself (`_entries`) in a local.

Reply:
I will push up another commit shortly to address this comment